### PR TITLE
feat: typescript scaffolding support

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

File: packages/init/templates/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    "esModuleInterop": true,
+    "declaration": true,
+    "forceConsistentCasingInFileNames": true,
+    "jsx": "react",
+    "lib": ["dom", "es2016", "es2017"],
+    "target": "esnext",
+    "module": "esnext",
+    "moduleResolution": "node",
+    "noFallthroughCasesInSwitch": true,
+    "noImplicitAny": false,
```
[the remark this answers is not in the document]
Comment (jquense) [reply packages/init/templates/tsconfig.json]: I think we should probably have it on right? Greenfield this is a better option I think

File: packages/init/templates/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "noImplicitReturns": true,
+    "noImplicitThis": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "strict": true,
+    "strictFunctionTypes": false,
+    "pretty": true,
```
[the remark this answers is not in the document]
Comment (jquense) [reply packages/init/templates/tsconfig.json]: remove

**Comment by itajaja** [comment]
let's roll with this, I'll try it on ONR

File: packages/init/templates/tsconfig.json.hbs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (taion) [inline packages/init/templates/tsconfig.json.hbs]: so this really isn't something we can share via e.g. a library?

Comment (itajaja) [reply packages/init/templates/tsconfig.json.hbs]: I think you can. you'd just reference the tsconfig in the `node_modules` folder

**Pull request merged**